lib/tools/system-calls.js

```diff
@@ -165,6 +165,31 @@ systemCallMethods.restartAdb = async function () {
   }
 };
 
+systemCallMethods.resetTelnetAuthToken = _.memoize(async function () {
+  // The methods is used to remove telnet auth token
+  // See http://tools.android.com/recent/emulator2516releasenotes for more details
+  const homeFolderPath = process.env[(process.platform === 'win32') ? 'USERPROFILE' : 'HOME'];
+  if (!homeFolderPath) {
+    log.warn('Cannot find the path to user home folder. Ignoring resetting of emulator\'s telnet authentication token');
+    return false;
+  }
+  const dstPath = path.resolve(homeFolderPath, '.emulator_console_auth_token');
+  log.debug(`Overriding ${dstPath} with an empty string to avoid telnet authentication for emulator commands`);
+  try {
+    await fs.writeFile(dstPath, '');
+  } catch (e) {
+    log.warn(`Error ${e.message} while resetting the content of ${dstPath}. Ignoring resetting of emulator\'s telnet authentication token`);
+    return false;
+  }
+  return true;
+});
+
+systemCallMethods.adbExecEmu = async function (cmd) {
+  await this.verifyEmulatorConnected();
+  await this.resetTelnetAuthToken();
```

Comment:
Nevermind - I was just thinking about creating a separate instance of memoized function. But this won't look good.